src/pages/ReportSettingsPage.js

```diff
@@ -110,8 +124,10 @@ class ReportSettingsPage extends Component {
 
     render() {
         const shouldShowRoomName = !ReportUtils.isPolicyExpenseChat(this.props.report);
-        const shouldDisableRename = ReportUtils.isDefaultRoom(this.props.report) || ReportUtils.isArchivedRoom(this.props.report);
         const linkedWorkspace = _.find(this.props.policies, policy => policy && policy.id === this.props.report.policyID);
+        const shouldDisableRename = ReportUtils.isDefaultRoom(this.props.report)
+            || ReportUtils.isArchivedRoom(this.props.report)
+            || this.shouldDisablePublicRoomRename(this.props.report, linkedWorkspace);
```

Comment:
why not move the first 2 parts into `shouldDisablePublicRoomRename` and just set this to the value returned from it?

```suggestion
        const shouldDisableRename = this.shouldDisablePublicRoomRename(this.props.report, linkedWorkspace)
```